File: src/main/java/work/lclpnet/ap2/impl/util/scoreboard/ScoreHandle.java
package work.lclpnet.ap2.impl.util.scoreboard;

import net.minecraft.scoreboard.number.NumberFormat;
import net.minecraft.text.Text;
import org.jetbrains.annotations.Nullable;
import work.lclpnet.kibu.translate.text.TextTranslatable;

public class ScoreHandle {

    private final String holder;
    private final TranslatedScoreboardObjective objective;

    public ScoreHandle(String holder, TranslatedScoreboardObjective objective) {
        this.holder = holder;
        this.objective = objective;
    }

    public void setScore(int score) {
        objective.setScore(holder, score);
    }

    public int getScore() {
        return objective.getScore(holder);
    }

    public void setDisplay(@Nullable Text text) {
        objective.setDisplayName(holder, text);
    }

    public void setDisplay(@Nullable TextTranslatable text) {
        objective.setDisplayName(holder, text);
    }

    public void setNumberFormat(NumberFormat numberFormat) {
        objective.setNumberFormat(holder, numberFormat);
    }
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/util/scoreboard/ScoreHandle.kt
package work.lclpnet.ap2.util.scoreboard

import net.minecraft.network.chat.Component
import net.minecraft.network.chat.numbers.NumberFormat

class ScoreHandle(
    val holder: String,
    val objective: CustomScoreboardObjective
) {

    fun setScore(score: Int) {
        objective.setScore(holder, score)
    }

    fun setDisplay(text: Component?) {
        objective.setDisplayName(holder, text)
    }

    fun setNumberFormat(numberFormat: NumberFormat?) {
        objective.setNumberFormat(holder, numberFormat)
    }
}
